File: redsheet-portal/src/app/database-models/arrival-and-opening-tactic.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectEventTimeline,  } from './'
export class ArrivalAndOpeningTactic {
    constructor(arrivalAndOpeningTactic?: ArrivalAndOpeningTactic )  {
        let properties = [ "arrivalAndOpeningTacticId", "arrivalAndOpeningTacticValue", "projectEventTimelineId",];
        for (let property of properties)
            if (arrivalAndOpeningTactic && arrivalAndOpeningTactic[property])
                this[property] = arrivalAndOpeningTactic[property];
    }
 
	arrivalAndOpeningTacticId : number =   0 ;
 
    @required()
    @maxLength(200)
	arrivalAndOpeningTacticValue : string =   undefined;
 
    @range(0,2147483647)
	projectEventTimelineId : number =   undefined;
	projectEventTimeline : ProjectEventTimeline  ;


}

File: RedSheet.DbEntities/Models/ArrivalAndOpeningTactic.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ArrivalAndOpeningTactics", Schema = "dbo")]
	[Serializable]
	public partial class ArrivalAndOpeningTactic  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ArrivalAndOpeningTacticId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ArrivalAndOpeningTacticValue { get; set; } 
 
	[RelationshipTableAttribue("ProjectEventTimelines","dbo")]
	//Relationships
	public int ProjectEventTimelineId { get; set; } 
 
	[ForeignKey("ProjectEventTimelineId")]
	public ProjectEventTimeline ProjectEventTimeline { get; set; } 

	public ArrivalAndOpeningTactic()
	{
    }

    }
}

